src/Strategy/AbstractCollectionStrategy.php

```diff
@@ -174,13 +133,8 @@ protected function getCollectionFromObjectByReference()
     /**
      * This method is used internally by array_udiff to check if two objects are equal, according to their
      * SPL hash. This is needed because the native array_diff only compare strings
-     *
-     * @param object $a
-     * @param object $b
-     *
-     * @return int
      */
-    protected function compareObjects($a, $b)
+    protected function compareObjects(object $a, object $b): int
     {
         return strcmp(spl_object_hash($a), spl_object_hash($b));
```

Comment:
Little detail: You could use a spaceship here.

Reply:
added in b652d2d